In [1]:
import seaborn as sns

from projet3 import utils
from projet3 import ead

/Users/alisaidomar/openclassroom_projet3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3249: DtypeWarning: Columns (0,24,25,26,28,44,49) have mixed types. Specify dtype option on import or set low_memory=False.
  if (await self.run_code(code, result,  async_=asy)):


# Qualité nutritionnel d'un repas à destination des sportifs. 
Pour un sportif être capable d'analyser la qualité de son repas et connaitre les
proportions des différent marconutriments.

## Pré-ambule
Depuis la source de données accessible à l'addresse https://world.openfoodfacts.org/data/data-fields.txt, nous allons réaliser les tâches suivantes:
1. Analyser la qualité de la données ; boulons, complétude.
2. Analyse statisque univarié.
3. Analyse statisque bivarié.
4. Exploitation des données dans le but d'extraire les proportions en terme de macronutriments.


### 1. Analyse qualitative de la donnée


Version

In [2]:
utils.print_version()

VERSION librairies
Python version 3.7.4 (v3.7.4:e09359112e, Jul  8 2019, 14:54:52) 
[Clang 6.0 (clang-600.0.57)]
Numpy version 1.17.3
Pandas version 0.25.2
Matplotlib version 3.1.1
Seaborn version 0.9.0


In [3]:
# Donnée brut
data_df = ead.get_data()

#Info
# les 10 premiere lignes


Extrait du dataset

In [4]:
data_df.head(10)

code                                                url  \
0         0000000000017  http://world-en.openfoodfacts.org/product/0000...   
1         0000000000031  http://world-en.openfoodfacts.org/product/0000...   
2  00000000001111111111  http://world-en.openfoodfacts.org/product/0000...   
3         0000000000123  http://world-en.openfoodfacts.org/product/0000...   
4         0000000000178  http://world-en.openfoodfacts.org/product/0000...   
5         0000000000208  http://world-en.openfoodfacts.org/product/0000...   
6         0000000000284  http://world-en.openfoodfacts.org/product/0000...   
7         0000000000291  http://world-en.openfoodfacts.org/product/0000...   
8         0000000000949  http://world-en.openfoodfacts.org/product/0000...   
9         0000000000970  http://world-en.openfoodfacts.org/product/0000...   

                      creator   created_t      created_datetime  \
0                     kiliweb  1529059080  2018-06-15T10:38:00Z   
1                    isagoofy  1539464774  2018-10-13T21:06:14Z   
2  openfoodfacts-contributors  1560020173  2019-06-08T18:56:13Z   
3                     kiliweb  1535737982  2018-08-31T17:53:02Z   
4                     kiliweb  1542456332  2018-11-17T12:05:32Z   
5                     kiliweb  1544207680  2018-12-07T18:34:40Z   
6                     kiliweb  1547120245  2019-01-10T11:37:25Z   
7                     kiliweb  1534239669  2018-08-14T09:41:09Z   
8                     kiliweb  1523440813  2018-04-11T10:00:13Z   
9                     kiliweb  1520506368  2018-03-08T10:52:48Z   

   last_modified_t last_modified_datetime                 product_name  \
0       1561463718   2019-06-25T11:55:18Z             Vitória crackers   
1       1539464817   2018-10-13T21:06:57Z                        Cacao   
2       1560020173   2019-06-08T18:56:13Z                      Sfiudwx   
3       1535737986   2018-08-31T17:53:06Z        Sauce Sweety chili 0%   
4       1542456333   2018-11-17T12:05:33Z                    Mini coco   
5       1544207683   2018-12-07T18:34:43Z        Pistou d'ail des ours   
6       1563551052   2019-07-19T15:44:12Z                    Pain maïs   
7       1534239732   2018-08-14T09:42:12Z                    Mendiants   
8       1565268412   2019-08-08T12:46:52Z    Salade de carottes râpées   
9       1561463726   2019-06-25T11:55:26Z  Fromage blanc aux myrtilles   

  generic_name quantity  ... carbon-footprint-from-meat-or-fish_100g  \
0          NaN      NaN  ...                                     NaN   
1          NaN    130 g  ...                                     NaN   
2          NaN    dgesc  ...                                     NaN   
3          NaN      NaN  ...                                     NaN   
4          NaN      NaN  ...                                     NaN   
5          NaN      NaN  ...                                     NaN   
6          NaN      NaN  ...                                     NaN   
7          NaN      NaN  ...                                     NaN   
8          NaN      NaN  ...                                     NaN   
9          NaN      NaN  ...                                     NaN   

  nutrition-score-fr_100g nutrition-score-uk_100g glycemic-index_100g  \
0                     NaN                     NaN                 NaN   
1                     NaN                     NaN                 NaN   
2                     NaN                     NaN                 NaN   
3                     NaN                     NaN                 NaN   
4                     NaN                     NaN                 NaN   
5                     NaN                     NaN                 NaN   
6                     NaN                     NaN                 NaN   
7                     NaN                     NaN                 NaN   
8                     1.0                     1.0                 NaN   
9                     NaN                     NaN                 NaN   

  water-ha

Info complète du dataset

In [5]:
data_df.info(verbose=True, null_counts=True)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009071 entries, 0 to 1009070
Data columns (total 175 columns):
code                                          1009071 non-null object
url                                           1009071 non-null object
creator                                       1009068 non-null object
created_t                                     1009071 non-null int64
created_datetime                              1009070 non-null object
last_modified_t                               1009071 non-null int64
last_modified_datetime                        1009071 non-null object
product_name                                  959819 non-null object
generic_name                                  92125 non-null object
quantity                                      316407 non-null object
packaging                                     186416 non-null object
packaging_tags                                186410 non-null object
brands                                        594578 n

Description

In [6]:
data_df.describe()


created_t  last_modified_t  cities  allergens_en  serving_quantity  \
count  1.009071e+06     1.009071e+06     0.0           0.0      2.777260e+05   
mean   1.518170e+09     1.540152e+09     NaN           NaN      4.000746e+16   
std    4.079759e+07     3.133868e+07     NaN           NaN      2.108382e+19   
min    0.000000e+00     1.333873e+09     NaN           NaN      0.000000e+00   
25%    1.489092e+09     1.525511e+09     NaN           NaN      2.800000e+01   
50%    1.524652e+09     1.552073e+09     NaN           NaN      4.900000e+01   
75%    1.550432e+09     1.563562e+09     NaN           NaN      1.130000e+02   
max    1.571790e+09     1.571790e+09     NaN           NaN      1.111111e+22   

       no_nutriments    additives_n  ingredients_from_palm_oil_n  \
count            0.0  447410.000000                447410.000000   
mean             NaN       1.622586                     0.027311   
std              NaN       2.349561                     0.164923   
min              NaN       0.000000                     0.000000   
25%              NaN       0.000000                     0.000000   
50%              NaN       1.000000                     0.000000   
75%              NaN       2.000000                     0.000000   
max              NaN      38.000000                     3.000000   

       ingredients_from_palm_oil  ingredients_that_may_be_from_palm_oil_n  \
count                        0.0                            447410.000000   
mean                         NaN                                 0.071889   
std                          NaN                                 0.316584   
min                          NaN                                 0.000000   
25%                          NaN                                 0.000000   
50%                          NaN                                 0.000000   
75%                          NaN                                 0.000000   
max                          NaN                                 6.000000   

       ...  carbon-footprint-from-meat-or-fish_100g  nutrition-score-fr_100g  \
count  ...                              9260.000000            286063.000000   
mean   ...                               631.841741                 9.217144   
std    ...                              6860.797317                 8.809736   
min    ...                                 0.049000               -15.000000   
25%    ...                               111.000000                 2.000000   
50%    ...                               334.985000                10.000000   
75%    ...                               617.125000                16.000000   
max    ...                            656298.600000                40.000000   

       nutrition-score-uk_100g  glycemic-index_100g  water-hardness_100g  \
count            286063.000000             3.000000                  0.0   
mean                  8.978092            29.333333                  NaN   
std                   9.272649            15.011107                  NaN   
min                 -15.000000            14.000000                  NaN   
25%                   1.000000            22.000000                  NaN   
50%                   9.000000            30.000000                  NaN   
75%                  17.000000            37.000000                  NaN   
max                  40.000000            44.000000                  NaN   

       choline_100g  phylloquinone_100g  beta-glucan_100g  inositol_100g  \
count     31.000000            4.000000          20.00000      37.000000   
mean       0.069577            0.090034           4.76000       0.026241   
std        0.031724            0.179977           4.24145       0.027702   
min        0.008000            0.000016           0.40000       0.001350   
25%        0.056500            0.000038           3.50000       0.006900   
50%        0.069000            0.000060           4.00000       0.022000   
75%        0.089000            0.090056     

Les lignes en doublons

In [7]:
data_df[data_df.duplicated()==True]

code                                                url  \
157221   254133007546  http://world-en.openfoodfacts.org/product/0254...   
371423  3217690055020  http://world-en.openfoodfacts.org/product/3217...   
472544   335820004060  http://world-en.openfoodfacts.org/product/3358...   
575022  3700225107238  http://world-en.openfoodfacts.org/product/3700...   
608928  3760124630473  http://world-en.openfoodfacts.org/product/3760...   
725712  5060136216664  http://world-en.openfoodfacts.org/product/5060...   
763343  5425021250251  http://world-en.openfoodfacts.org/product/5425...   
821512  7613033065569  http://world-en.openfoodfacts.org/product/7613...   
971452  8711743560042  http://world-en.openfoodfacts.org/product/8711...   
977400  8715061627104  http://world-en.openfoodfacts.org/product/8715...   
979569  8717703720921  http://world-en.openfoodfacts.org/product/8717...   
980460  8718215382959  http://world-en.openfoodfacts.org/product/8718...   
981728  8718546163784  http://world-en.openfoodfacts.org/product/8718...   
985141  8722100046345  http://world-en.openfoodfacts.org/product/8722...   

                           creator   created_t      created_datetime  \
157221  openfoodfacts-contributors  1537196361  2018-09-17T14:59:21Z   
371423                     kiliweb  1554670701  2019-04-07T20:58:21Z   
472544  openfoodfacts-contributors  1563792647  2019-07-22T10:50:47Z   
575022                     kiliweb  1558182525  2019-05-18T12:28:45Z   
608928  openfoodfacts-contributors  1552607164  2019-03-14T23:46:04Z   
725712                    sebleouf  1556491987  2019-04-28T22:53:07Z   
763343                     kiliweb  1560100380  2019-06-09T17:13:00Z   
821512  openfoodfacts-contributors  1490187090  2017-03-22T12:51:30Z   
971452                    sebleouf  1556491307  2019-04-28T22:41:47Z   
977400                    sebleouf  1556478017  2019-04-28T19:00:17Z   
979569                    sebleouf  1556474386  2019-04-28T17:59:46Z   
980460                    sebleouf  1556496872  2019-04-29T00:14:32Z   
981728                    teolemon  1544099493  2018-12-06T12:31:33Z   
985141                    sebleouf  1556447499  2019-04-28T10:31:39Z   

        last_modified_t last_modified_datetime  \
157221       1556215968   2019-04-25T18:12:48Z   
371423       1556111603   2019-04-24T13:13:23Z   
472544       1571503410   2019-10-19T16:43:30Z   
575022       1558183963   2019-05-18T12:52:43Z   
608928       1556658319   2019-04-30T21:05:19Z   
725712       1556573491   2019-04-29T21:31:31Z   
763343       1563048678   2019-07-13T20:11:18Z   
821512       1556217855   2019-04-25T18:44:15Z   
971452       1556573426   2019-04-29T21:30:26Z   
977400       1556573120   2019-04-29T21:25:20Z   
979569       1556573310   2019-04-29T21:28:30Z   
980460       1556571364   2019-04-29T20:56:04Z   
981728       1544099563   2018-12-06T12:32:43Z   
985141       1556658133   2019-04-30T21:02:13Z   

                            product_name generic_name    quantity  ...  \
157221                        pain rando          NaN         NaN  ...   
371423                 Syrup green apple          NaN         NaN  ...   
472544                               NaN          NaN       500 g  ...   
575022                  Lentilles corail          NaN       500 g  ...   
608928                               NaN          NaN         NaN  ...   
725712                               NaN          NaN         NaN  ...   
763343  Filet de poulet sauce Chakchouka          NaN         NaN  ...   
821512                               NaN          NaN         NaN  ...   
971452                               NaN          NaN  18 x 1.5 g  ...   
977400                               NaN          NaN         NaN  ...   
979569                               NaN          NaN         NaN  ...   
980460                               NaN          NaN         NaN  ...   
981728                               NaN          NaN         NaN  ...   
985141  

Complétude des colonnes

In [8]:
completude_df = ead.get_columns_completude()

In [9]:
completude_gt_70 = completude_df[completude_df["Completude"] >= 70.0]

In [10]:
completude_gt_70

Column  Completude
0                      code  100.000000
1                       url  100.000000
2                   creator   99.999703
3                 created_t  100.000000
4          created_datetime   99.999901
5           last_modified_t  100.000000
6    last_modified_datetime  100.000000
7              product_name   95.119075
31                countries   99.827663
32           countries_tags   99.827564
33             countries_en   99.827564
55            pnns_groups_1   98.951412
56            pnns_groups_2   99.962143
57                   states  100.000000
58              states_tags  100.000000
59                states_en  100.000000
62                image_url   73.650913
63          image_small_url   73.650913
68              energy_100g   80.192771
70                 fat_100g   79.722339
71       saturated-fat_100g   76.786569
106      carbohydrates_100g   79.718077
107             sugars_100g   78.328780
117           proteins_100g   79.911721
121               salt_100g   77.946646
122             sodium_100g   77.941592

In [11]:
completude_gt_70.shape

(26, 2)

In [12]:
completude_lt_001 = completude_df[completude_df["Completude"] < 0.01]

In [13]:
completude_lt_001

Column  Completude
27                                  cities    0.000000
36                            allergens_en    0.000000
42                           no_nutriments    0.000000
44                               additives    0.002973
48               ingredients_from_palm_oil    0.000000
51   ingredients_that_may_be_from_palm_oil    0.000000
72                      -butyric-acid_100g    0.000099
73                      -caproic-acid_100g    0.000000
74                     -caprylic-acid_100g    0.000099
75                       -capric-acid_100g    0.000396
76                       -lauric-acid_100g    0.000694
77                     -myristic-acid_100g    0.000099
78                     -palmitic-acid_100g    0.000297
79                      -stearic-acid_100g    0.000099
80                    -arachidic-acid_100g    0.000198
81                      -behenic-acid_100g    0.000099
82                   -lignoceric-acid_100g    0.000000
83                      -cerotic-acid_100g    0.000297
84                     -montanic-acid_100g    0.000099
85                     -melissic-acid_100g    0.000000
90             -eicosapentaenoic-acid_100g    0.009712
94                  -arachidonic-acid_100g    0.001487
95              -gamma-linolenic-acid_100g    0.000297
96       -dihomo-gamma-linolenic-acid_100g    0.000099
97                        omega-9-fat_100g    0.003865
98                        -oleic-acid_100g    0.002081
99                      -elaidic-acid_100g    0.000000
100                     -gondoic-acid_100g    0.000000
101                        -mead-acid_100g    0.000000
102                      -erucic-acid_100g    0.000000
103                    -nervonic-acid_100g    0.000000
109                          -glucose_100g    0.004757
110                         -fructose_100g    0.006739
112                          -maltose_100g    0.000892
113                    -maltodextrins_100g    0.001387
118                            casein_100g    0.003766
119                    serum-proteins_100g    0.003369
120                       nucleotides_100g    0.001189
125                     beta-carotene_100g    0.006838
139                            silica_100g    0.009910
152                          chromium_100g    0.005748
153                        molybdenum_100g    0.005946
156                           taurine_100g    0.009117
163                        chlorophyl_100g    0.000198
168                    glycemic-index_100g    0.000297
169                    water-hardness_100g    0.000000
170                           choline_100g    0.003072
171                     phylloquinone_100g    0.000396
172                       beta-glucan_100g    0.001982
173                          inositol_100g    0.003667
174                         carnitine_100g    0.001487

In [14]:
completude_lt_001.shape

(51, 2)

#### Résultat
Données issues de l'open data disponible gratuitement. Elle est composé de plus d'un million d'entrée et 175 colonnes:
- pour 51 colonnes on dispose de moins des 0.01% des lignes de renseignées, ces colonnes peuvent être omises de notre analyse
- pour 26 colonnes on dispose de plus de 70% des lignes de renseignées
- 14 lignes sont en doublons
- 119 colonnes correspondes à des données quantitative dont seulement 50 pour lesquels on dispose de plus de 1000 lignes (0.01% du volumes) de renseignées

#### Conclusion
Les premières analyses sont plutôt positives; on dispose de peu de doublons et on dispose de 26 colonnes pour lesquelles plus de 70% des lignes sont renseignées.


### 2. Analyse statisque univariée
Dans cette partie nous allons procéder à l'analyse univariée sur les colonnes retenues dans le cadre de notre analyse

#### Choix des colonnes
Dans un premier temps, sont retenues les colonnes pour lesquelles 70% de lignes sont remplies

In [15]:
completude_gt_70["Column"]

0                        code
1                         url
2                     creator
3                   created_t
4            created_datetime
5             last_modified_t
6      last_modified_datetime
7                product_name
31                  countries
32             countries_tags
33               countries_en
55              pnns_groups_1
56              pnns_groups_2
57                     states
58                states_tags
59                  states_en
62                  image_url
63            image_small_url
68                energy_100g
70                   fat_100g
71         saturated-fat_100g
106        carbohydrates_100g
107               sugars_100g
117             proteins_100g
121                 salt_100g
122               sodium_100g
Name: Column, dtype: object

Seules les colonnes
```
68                energy_100g
70                   fat_100g
71         saturated-fat_100g
106        carbohydrates_100g
107               sugars_100g
117             proteins_100g
121                 salt_100g
122               sodium_100g
```
Correspondes à des données quantitatives. Nous allons donc étendre notre analyse a certaines colonnes incluses dans

In [16]:
completude_df[(completude_df["Completude"] >= 10.0)&(completude_df["Completude"] < 70.0)]

Column  Completude
9                                   quantity   31.356267
10                                 packaging   18.474022
11                            packaging_tags   18.473428
12                                    brands   58.923307
13                               brands_tags   58.918451
14                                categories   36.906323
15                           categories_tags   36.905926
16                             categories_en   36.905926
21                                    labels   19.570972
22                               labels_tags   19.573350
23                                 labels_en   19.573350
29                           purchase_places   12.146222
30                                    stores   17.810937
34                          ingredients_text   44.350893
40                              serving_size   27.511444
41                          serving_quantity   27.522939
43                               additives_n   44.338803
45                            additives_tags   24.136855
46                              additives_en   24.136855
47               ingredients_from_palm_oil_n   44.338803
50   ingredients_that_may_be_from_palm_oil_n   44.338803
53                        nutrition_grade_fr   28.349244
54                                nova_group   31.857818
60                             main_category   36.899187
61                          main_category_en   36.899187
64                     image_ingredients_url   57.696733
65               image_ingredients_small_url   57.696733
66                       image_nutrition_url   32.919686
67                 image_nutrition_small_url   32.919686
104                           trans-fat_100g   14.303652
105                         cholesterol_100g   14.382140
116                               fiber_100g   29.860238
124                           vitamin-a_100g   13.779704
129                           vitamin-c_100g   14.132801
143                             calcium_100g   14.473610
145                                iron_100g   14.142711
166                  nutrition-score-fr_100g   28.349145
167                  nutrition-score-uk_100g   28.349145

On retient alors les colonnes suivantes
```
104	trans-fat_100g	14.303652
105	cholesterol_100g	14.382140
116	fiber_100g	29.860238
124	vitamin-a_100g	13.779704
129	vitamin-c_100g	14.132801
143	calcium_100g	14.473610
145	iron_100g	14.142711
```
On regrettera de ne pas pouvoir retenir les vitamines B. 

In [17]:
all_columns = [
    'calcium_100g','cholesterol_100g', 'trans-fat_100g', 'iron_100g',
    'vitamin-c_100g', 'vitamin-a_100g', 'fat_100g', 'saturated-fat_100g', 
    'sugars_100g', 'carbohydrates_100g', 'proteins_100g','salt_100g',  
    'sodium_100g', 'energy_100g'
]

Données abérante

In [18]:
data_restricted_df = data_df[all_columns]
data_restricted_df.describe()

calcium_100g  cholesterol_100g  trans-fat_100g      iron_100g  \
count  146049.000000     145126.000000   144334.000000  142710.000000   
mean        0.214219          0.028468        0.073794       0.006987   
std         6.556936          0.901867        1.494106       0.339303   
min         0.000000          0.000000       -3.570000      -0.000260   
25%         0.000000          0.000000        0.000000       0.000000   
50%         0.036000          0.000000        0.000000       0.001030   
75%         0.114000          0.020000        0.000000       0.002480   
max       950.000000        141.000000      369.000000      50.000000   

       vitamin-c_100g  vitamin-a_100g      fat_100g  saturated-fat_100g  \
count   142610.000000   139047.000000  8.044550e+05        7.748310e+05   
mean         0.038394        0.003074  2.495856e+01        1.111774e+01   
std          2.763584        0.454334  9.756626e+03        5.099556e+03   
min         -0.002100       -0.000340  0.000000e+00        0.000000e+00   
25%          0.000000        0.000000  6.000000e-01        1.000000e-01   
50%          0.000000        0.000000  6.800000e+00        1.900000e+00   
75%          0.004000        0.000107  2.140000e+01        7.400000e+00   
max        716.981100      100.000000  8.748748e+06        4.488758e+06   

        sugars_100g  carbohydrates_100g  proteins_100g     salt_100g  \
count  7.903930e+05       804412.000000   8.063660e+05  7.865370e+05   
mean   2.002262e+01           29.904906   1.185912e+02  1.055407e+02   
std    5.146044e+03          961.252502   9.889830e+04  9.120719e+04   
min   -1.786000e+01            0.000000  -8.000000e+02  0.000000e+00   
25%    8.000000e-01            3.570000   1.200000e+00  6.000000e-02   
50%    4.000000e+00           15.400000   5.700000e+00  5.200000e-01   
75%    2.000000e+01           54.000000   1.200000e+01  1.343660e+00   
max    4.574988e+06       857855.000000   8.880858e+07  8.088869e+07   

        sodium_100g   energy_100g  
count  7.864860e+05  8.092020e+05  
mean   4.241152e+01  8.237200e+36  
std    3.648451e+04  7.409827e+39  
min    0.000000e+00  0.000000e+00  
25%    2.000000e-02  4.300000e+02  
50%    2.001520e-01  1.096000e+03  
75%    5.280000e-01  1.674000e+03  
max    3.235548e+07  6.665559e+42

On constate des données < 0 et d'autre > 100g, on va donc procéder à des remplacements

In [19]:
data_restricted_df[data_restricted_df <= 0] = 0
data_restricted_df[data_restricted_df >= 100] = 100
data_restricted_df[data_restricted_df.isnull()] = 0

/Users/alisaidomar/openclassroom_projet3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/Users/alisaidomar/openclassroom_projet3/lib/python3.7/site-packages/pandas/core/frame.py:3530: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._where(-key, value, inplace=True)
/Users/alisaidomar/openclassroom_projet3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[r

In [20]:
# on check 
data_restricted_df.describe()

calcium_100g  cholesterol_100g  trans-fat_100g     iron_100g  \
count  1.009071e+06      1.009071e+06    1.009071e+06  1.009071e+06   
mean   2.221523e-02      3.998166e-03    1.022747e-02  9.881689e-04   
std    8.343660e-01      3.076227e-01    4.232882e-01  1.276237e-01   
min    0.000000e+00      0.000000e+00    0.000000e+00  0.000000e+00   
25%    0.000000e+00      0.000000e+00    0.000000e+00  0.000000e+00   
50%    0.000000e+00      0.000000e+00    0.000000e+00  0.000000e+00   
75%    0.000000e+00      0.000000e+00    0.000000e+00  0.000000e+00   
max    1.000000e+02      1.000000e+02    1.000000e+02  5.000000e+01   

       vitamin-c_100g  vitamin-a_100g      fat_100g  saturated-fat_100g  \
count    1.009071e+06    1.009071e+06  1.009071e+06        1.009071e+06   
mean     3.906689e-03    4.236117e-04  1.080204e+01        4.002698e+00   
std      3.816272e-01    1.686563e-01  1.658136e+01        7.289218e+00   
min      0.000000e+00    0.000000e+00  0.000000e+00        0.000000e+00   
25%      0.000000e+00    0.000000e+00  0.000000e+00        0.000000e+00   
50%      0.000000e+00    0.000000e+00  2.680000e+00        5.000000e-01   
75%      0.000000e+00    0.000000e+00  1.700000e+01        5.000000e+00   
max      1.000000e+02    1.000000e+02  1.000000e+02        1.000000e+02   

        sugars_100g  carbohydrates_100g  proteins_100g     salt_100g  \
count  1.009071e+06        1.009071e+06   1.009071e+06  1.009071e+06   
mean   1.108244e+01        2.274243e+01   6.618043e+00  1.166351e+00   
std    1.882548e+01        2.772386e+01   9.187173e+00  5.486861e+00   
min    0.000000e+00        0.000000e+00   0.000000e+00  0.000000e+00   
25%    0.000000e+00        0.000000e+00   0.000000e+00  0.000000e+00   
50%    1.900000e+00        8.450000e+00   3.390000e+00  1.500000e-01   
75%    1.230000e+01        4.600000e+01   9.500000e+00  1.089660e+00   
max    1.000000e+02        1.000000e+02   1.000000e+02  1.000000e+02   

        sodium_100g   energy_100g  
count  1.009071e+06  1.009071e+06  
mean   4.962576e-01  7.657456e+01  
std    3.050819e+00  4.181017e+01  
min    0.000000e+00  0.000000e+00  
25%    0.000000e+00  9.900000e+01  
50%    5.600000e-02  1.000000e+02  
75%    4.280000e-01  1.000000e+02  
max    1.000000e+02  1.000000e+02

In [21]:
for c in data_restricted_df.columns:
    ax = sns.boxplot(x=c, data=data_restricted_df[c])


ValueError: Could not interpret input 'calcium_100g'